src/test/poll/test_poll.rs

```diff
@@ -0,0 +1,224 @@
+/*
+ * The Shadow Simulator
+ * See LICENSE for licensing information
+ */
+
+use test_utils::set;
+use test_utils::TestEnvironment as TestEnv;
+
+fn test_fd_write(fd: i32) -> i32 {
+    return unsafe { libc::write(fd, "test".as_ptr() as *const libc::c_void, 4) as i32 };
+}
+
+fn test_fd_read_cmp(fd: i32) -> Result<(), String> {
+    let mut buf = [0_u8; 5];
+    unsafe { libc::read(fd, buf.as_mut_ptr() as *mut libc::c_void, 4) };
+    println!("read from buf: {:?}", buf);
+    let rv = unsafe {
+        libc::strncmp(
+            buf.as_ptr() as *const libc::c_char,
+            "test".as_ptr() as *const libc::c_char,
+            4,
+        )
+    };
+    if rv != 0 {
+        return Err("error: did not read 'test' from pipe.".to_string());
+    } else {
+        return Ok(());
+    }
```

Comment:
Normally we try to avoid adding rust dependencies, but we're already using the nix crate in shadow, so we want to start using it tests as well for syscalls that we're not explicitly testing to make the code easier to write and read.

Reply:
> Normally we try to avoid adding rust dependencies, but we're already using the nix crate in shadow, so we want to start using it tests as well for syscalls that we're not explicitly testing to make the code easier to write and read.

Saw use of `nix` when reviewing code, and agree using it is a great idea. Will add it with the fixups for this PR.